### DISCUSSION TITLE
What is the recommended way to sync external values into machine context?

### DISCUSSION THREAD

**Discussion by wuweiweiwu** [statelyai/xstate#2031] [Q&A] [answered]
xstate react `useMachine` accepts an initial context but what is the recommended way to sync some external value (eg: react state value) into context but also preserving the current state of the invoked machine?

Should I be using `machine.withContext`? and storing the instance of the machine in react state?

Or should I define an action in `useMachine` that handles the syncing and use `useEffect` to send a sync event that copies that value into context?

example: https://codesandbox.io/s/vigorous-cori-urixg?file=/src/index.js

**Comment by davidkpiano** [comment, marked as answer]
> Or should I define an action in `useMachine` that handles the syncing and use `useEffect` to send a sync event that copies that value into context?

Yes, this is what you should be doing:

```js
useEffect(() => {
  send({ type: 'someUpdateEvent', value });
}, [value]);
```
